aio_pika/abc.py

```diff
@@ -0,0 +1,658 @@
+import asyncio
+from abc import ABC, abstractmethod, abstractproperty
+from datetime import datetime, timedelta
+from enum import Enum, IntEnum, unique
+from types import TracebackType
+from typing import (
+    Any, AsyncContextManager, Awaitable, Callable, Dict, FrozenSet, Generator,
+    Iterable, MutableMapping, NamedTuple, Optional, Set, Tuple, Type, TypeVar,
+    Union,
+)
+
+import aiormq
+from aiormq.abc import ExceptionType
+from pamqp.common import Arguments
+from yarl import URL
+
+from .pool import PoolInstance
+from .tools import CallbackCollection
+
+
+TimeoutType = Optional[Union[int, float]]
+
+NoneType = type(None)
+DateType = Union[int, datetime, float, timedelta, None]
+ExchangeParamType = Union["AbstractExchange", str]
+ConsumerTag = str
+
+MILLISECONDS = 1000
+ZERO_TIME = datetime.utcfromtimestamp(0)
+
+
+@unique
+class ExchangeType(str, Enum):
+    FANOUT = "fanout"
+    DIRECT = "direct"
+    TOPIC = "topic"
+    HEADERS = "headers"
+    X_DELAYED_MESSAGE = "x-delayed-message"
+    X_CONSISTENT_HASH = "x-consistent-hash"
+    X_MODULUS_HASH = "x-modulus-hash"
+
+
+@unique
+class DeliveryMode(IntEnum):
+    NOT_PERSISTENT = 1
+    PERSISTENT = 2
+
+
+@unique
+class TransactionStates(str, Enum):
+    created = "created"
+    commited = "commited"
+    rolled_back = "rolled back"
+    started = "started"
+
+
+class DeclarationResult(NamedTuple):
+    message_count: int
+    consumer_count: int
+
+
+class AbstractTransaction:
+    state: TransactionStates
+
+    @abstractproperty
+    def channel(self) -> "AbstractChannel":
```

Comment:
`-> aiormq.abc.AbstractChannel:`

Reply:
Nope, Transaction returns aio-pike's Channel instance. aiormq's channel is AMQP transport object and not exposes from aio-pika's classes.